### PR TITLE
feat: update cheatcodes-test CI to use local era-test-node

### PR DESCRIPTION
**Pull request by aon**
# What :computer: 
* Update cheatcodes-test CI to use local era-test-node

# Why :hand:
* To fix rate limit issue caused when executing cheatcodes-test CI due to rate limiting by mainnet RPC

# Evidence :camera:
CI run.

### CODE REVIEW

File: crates/era-cheatcodes/tests/foundry.toml [hunk 1]
```diff
@@ -5,6 +5,6 @@ libs = ['lib']
 
 [rpc_endpoints]
 local = "${ERA_TEST_NODE_RPC_URL}"
-mainnet = "https://mainnet.era.zksync.io:443"
+mainnet = "${RPC_MAINNET}"
```
Comment (dutterbutter) [inline crates/era-cheatcodes/tests/foundry.toml:8]: Can we change this variable to be `local`? Just so its consistent with using era-test-node.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -115,15 +115,33 @@ jobs:
...
+            mode: fork
+            network: mainnet
+            log: info
+            logFilePath: era_test_node.log
+            target: x86_64-unknown-linux-gnu
+            releaseTag: latest
+
         - name: Test Cheatcodes
+          env:
+            RPC_MAINNET: http://localhost:8011
```
Comment (dutterbutter) [inline .github/workflows/test.yml:140]: Can we change this to `RPC_LOCAL`

File: crates/era-cheatcodes/tests/src/cheatcodes/RpcUrls.t.sol [hunk 3]
```diff
@@ -10,9 +10,7 @@ contract RpcUrlsTest is Test {
         string memory rpc_url = vm.rpcUrl("mainnet");
         require(
             keccak256(bytes(rpc_url)) ==
-                keccak256(
-                    abi.encodePacked("https://mainnet.era.zksync.io:443")
-                ),
+                keccak256(abi.encodePacked(vm.envString("RPC_MAINNET"))),
```
Comment (dutterbutter) [inline crates/era-cheatcodes/tests/src/cheatcodes/RpcUrls.t.sol:13]: Same as above

File: crates/era-cheatcodes/tests/src/cheatcodes/RpcUrls.t.sol [hunk 4]
```diff
@@ -34,7 +32,7 @@ contract RpcUrlsTest is Test {
         );
         require(
             keccak256(bytes(rpc_urls[1][1])) ==
-                keccak256("https://mainnet.era.zksync.io:443"),
+                keccak256(bytes(vm.envString("RPC_MAINNET"))),
```
Comment (dutterbutter) [inline crates/era-cheatcodes/tests/src/cheatcodes/RpcUrls.t.sol:35]: Same as above

**Review by dutterbutter — CHANGES_REQUESTED** [review]

File: crates/era-cheatcodes/tests/foundry.toml (hunk 1 above)
Comment (aon) [reply crates/era-cheatcodes/tests/foundry.toml:8]: Sure, I'll make the change . FYI this involves changing all references in tests (such as https://github.com/matter-labs/foundry-zksync/blob/eb2ae26710a124f3e66bf0cb52b41d335cf20d9b/crates/era-cheatcodes/tests/src/cheatcodes/Fork.t.sol#L25) to reference the "local" endpoint.

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**